### PR TITLE
Various bug fixes

### PR DESCRIPTION
**Pull request by lynn**
* Disallow duplicate enum members `enum E { A=1; A=2; }`.
* Disallow duplicate discriminators `union U { 1 -> struct V{}; 1 -> struct W{} }`.
* Generate doc comments for union branches properly.
* Generate `bebop.hpp` alongside C++ output.
* C++: Use `::bebop` so that generated code works in a namespace.
* Report unknown command line flags.
* Report when `file.json` does not exist in `--config file.json`.

### CODE REVIEW

File: Core/Meta/BebopSchema.cs [hunk 1]
```diff
@@ -83,14 +83,20 @@ public void Validate()
                 }
                 if (definition is EnumDefinition ed)
                 {
-                    HashSet<uint> seen = new HashSet<uint>();
+                    HashSet<uint> values = new HashSet<uint>();
+                    HashSet<string> names = new HashSet<string>();
```
Comment (AndrewMD5) [inline Core/Meta/BebopSchema.cs:87]: You'll want this HashSet to be case-insensitive
```new HashSet<string>(StringComparer.OrdinalIgnoreCase);```

File: Laboratory/C++/gen/jazz.hpp [hunk 2]
```diff
@@ -17,25 +18,28 @@ struct Musician {
...
+  static Musician decode(const uint8_t *sourceBuffer) {
     Musician result;
-    bebop::BebopReader reader{buffer};
-    Musician::decodeInto(result, reader);
+    Musician::decodeInto(sourceBuffer, result);
     return result;
   }
 
-  static void decodeInto(Musician& target, bebop::BebopReader& reader) {
+  static void decodeInto(const uint8_t *sourceBuffer, Musician& target) {
```
Comment (apecoraro) [inline Laboratory/C++/gen/jazz.hpp:37]: Minor nit pick is that you have two different styles for `*` and `&` here (right align and left align). I would expect either:
```C++
static void decodeInto(const uint8_t* sourceBuffer, Musician& target)
```
or
```C++
static void decodeInto(const uint8_t *sourceBuffer, Musician &target)
```

File: Laboratory/C++/gen/jazz.hpp [hunk 3]
```diff
@@ -46,13 +50,12 @@ struct Song {
   std::optional<uint16_t> year;
   std::optional<std::vector<Musician>> performers;
 
-  static std::unique_ptr<std::vector<uint8_t>> encode(const Song& message) {
-    bebop::BebopWriter writer{};
+  static void encodeInto(const Song& message, std::vector<uint8_t> &targetBuffer) {
```
Comment (apecoraro) [inline Laboratory/C++/gen/jazz.hpp:53]: Same nit pick here.

File: Laboratory/C++/gen/jazz.hpp [hunk 4]
```diff
@@ -78,14 +81,18 @@ struct Song {
     writer.fillMessageLength(pos, end - start);
   }
 
-  static Song decode(const uint8_t *buffer) {
+  static Song decode(const uint8_t *sourceBuffer) {
```
Comment (apecoraro) [inline Laboratory/C++/gen/jazz.hpp:84]: Whatever style you chooose, reminder to use it here too.

**Review by apecoraro — COMMENTED** [review]
The changes to the code generation look good to me.

**Review by AndrewMD5 — APPROVED** [review]

**Pull request merged**